src/Process.php

```diff
@@ -66,17 +66,22 @@ public function start(): Promise {
         $this->resolved = new Deferred();
 
         return call(function () {
-            if (!$this->open) {
-                yield $this->session->open();
-
-                $this->open = true;
-            }
-
-            foreach ($this->env as $key => $value) {
-                yield $this->session->env($key, $value);
-            }
-
-            yield $this->session->exec($this->command);
+        	try {
+				if (!$this->open) {
+					yield $this->session->open();
+
+					$this->open = true;
+				}
+
+				foreach ($this->env as $key => $value) {
+					yield $this->session->env($key, $value);
+				}
+
+				yield $this->session->exec($this->command);
+			} catch(\Exception $exception) {
+        		$this->resolved = null;
+        		throw $exception;
+			}
```

Comment:
Bad indentation

Reply:
I want to use cs-fix for that but need to reformat master first.